exchange/bidder.go

```diff
@@ -718,3 +724,28 @@ func getBidTypeForAdjustments(bidType openrtb_ext.BidType, impID string, imp []o
 	}
 	return string(bidType)
 }
+
+type bidderTmaxContext interface {
+	Deadline() (deadline time.Time, ok bool)
+	RemainingDurationMS(deadline time.Time) int64
+}
+type bidderTmaxCtx struct{ ctx context.Context }
+
+func (b *bidderTmaxCtx) RemainingDurationMS(deadline time.Time) int64 {
+	return time.Until(deadline).Milliseconds()
+}
+func (b *bidderTmaxCtx) Deadline() (deadline time.Time, ok bool) {
+	deadline, ok = b.ctx.Deadline()
+	return
+}
+
+func getBidderTmax(ctx bidderTmaxContext, requestTmaxMS int64, tmaxAdjustments config.TmaxAdjustments) int64 {
+	if tmaxAdjustments.Enabled &&
+		tmaxAdjustments.BidderResponseDurationMin != 0 &&
+		(tmaxAdjustments.BidderNetworkLatencyBuffer != 0 || tmaxAdjustments.PBSResponsePreparationDuration != 0) {
```

Comment:
This block of code looks like a good candidate to be evaluated once at the server start up. These parameters come with the config and will not change per-request.

For the `exchange.AuctionRequest` you may use another, more convenient structure with some pre-processed values from configs.

Another point toward this idea is that you use same exact structure you used for config representation. 
These structs should be separated (model-view-presenter design pattern). Look at the `Activities` example in `exchange.AuctionRequest`.

Reply:
>This block of code looks like a good candidate to be evaluated once at the server start up. These parameters come with the config and will not change per-request.

`0` is default value   for `BidderResponseDurationMin, .BidderNetworkLatencyBuffer and .PBSResponsePreparationDuration`.  So we cannot add them at server startup or validate them to return error if 0.  

By checking these values `!=0`, we make sure that deadline from context are extracted and reduced only if needed. This saves us with some time

Reply:
Sorry, this is not what I mean, I didn't express myself clear. My general proposal here is to preprocess data from deps.cfg.TmaxAdjustments and save it to a new structure that will be passed to the endpoints.
Here is a draft with changes: https://github.com/VeronikaSolovei9/prebid-server/pull/4 
Please see my commit on top of your changes: https://github.com/VeronikaSolovei9/prebid-server/pull/4/commits/99653260ba2f5a1b378a46dfb0bfebe424f18f9a

Reply:
@VeronikaSolovei9 thanks for the draft. For now I will defer to using `config.TmaxAdjustments` struct. Lets build out this feature and then we can do this enhancement.  Being said that, lets see what other reviewers has feedback on this

Reply:
Why do you want to defer it? This change is simple and makes sense, it is more optimal than current approach and it's all done in draft so you see how it works already. It's better to do everything right at the beginning (especially simple things like this), rather then do it in the follow up PR that can possibly never happen or will be forgotten.

Reply:
@VeronikaSolovei9  https://github.com/prebid/prebid-server/pull/2930/commits/fb19f5c1af6a39a55aa6c564f1fac7e4802908e0 makes the changes